File: pyNCAA/scripts/_test_pcover.py
#!/usr/bin/env python3
# scripts/_test_pcover.py — Test P(cover) thresholds
import json
import re

h = json.loads(open("data/history.json").read())
thresholds = [0.57, 0.58, 0.59, 0.60, 0.61, 0.62, 0.63, 0.64, 0.65, 0.67, 0.70]

print("=== NCAA: Testing P(cover) thresholds ===")
print("Thresh | W    | L    | Win%  | Units   | Picks/day | Fav W% | Dog W%")
print("-------|------|------|-------|---------|-----------|--------|-------")

for t in thresholds:
    w, l = 0, 0
    fav_w, fav_l, dog_w, dog_l = 0, 0, 0, 0
    for r in (h.get("runs") or []):
        if r.get("burnIn"): continue
        for g in (r.get("games") or []):
            if not g.get("sPick") or g["sPick"] == "PASS": continue
            if not g.get("sResult") or not g.get("pCover"): continue
            if g["pCover"] < t: continue
            m = re.search(r"([+-])(\d+(?:\.\d+)?)$", g["sPick"])
            is_fav = m and m.group(1) == "-"
            if g["sResult"] == "WIN":
                w += 1
                if is_fav: fav_w += 1
                else: dog_w += 1
            elif g["sResult"] == "LOSS":
                l += 1
                if is_fav: fav_l += 1
                else: dog_l += 1
    n = w + l
    pct = f"{w / n * 100:.1f}" if n > 0 else "N/A"
    units = f"{(w - l * 1.1):.1f}"
    per_day = f"{n / 73:.1f}"
    fav_pct = f"{fav_w / (fav_w + fav_l) * 100:.1f}" if (fav_w + fav_l) > 0 else "N/A"
    dog_pct = f"{dog_w / (dog_w + dog_l) * 100:.1f}" if (dog_w + dog_l) > 0 else "N/A"
    mark = " <- current" if t == 0.63 else ""
    print(f"{t:.2f}".ljust(7) + f"| {w!s:<5}| {l!s:<5}| {pct:<6}| {units:<8}| {per_day:<10}| {fav_pct:<7}| {dog_pct}{mark}")

File: jsNCAA/scripts/_test_pcover.mjs
import fs from 'fs';
const h = JSON.parse(fs.readFileSync('data/history.json','utf8'));

// Test different pCover thresholds on existing graded picks
const thresholds = [0.57, 0.58, 0.59, 0.60, 0.61, 0.62, 0.63, 0.64, 0.65, 0.67, 0.70];

console.log('=== NCAA: Testing P(cover) thresholds ===');
console.log('Thresh | W    | L    | Win%  | Units   | Picks/day | Fav W% | Dog W%');
console.log('-------|------|------|-------|---------|-----------|--------|-------');

for (const t of thresholds) {
  let w = 0, l = 0, favW = 0, favL = 0, dogW = 0, dogL = 0;

  for (const r of (h.runs || [])) {
    if (r.burnIn) continue;
    for (const g of (r.games || [])) {
      if (!g.sPick || g.sPick === 'PASS') continue;
      if (!g.sResult || !g.pCover) continue;
      if (g.pCover < t) continue;

      const m = g.sPick.match(/([+-])(\d+(?:\.\d+)?)$/);
      const isFav = m && m[1] === '-';

      if (g.sResult === 'WIN') {
        w++;
        if (isFav) favW++; else dogW++;
      } else if (g.sResult === 'LOSS') {
        l++;
        if (isFav) favL++; else dogL++;
      }
    }
  }

  const n = w + l;
  const pct = n > 0 ? (w / n * 100).toFixed(1) : 'N/A';
  const units = (w - l * 1.1).toFixed(1);
  const perDay = (n / 73).toFixed(1);
  const favPct = (favW + favL) > 0 ? (favW / (favW + favL) * 100).toFixed(1) : 'N/A';
  const dogPct = (dogW + dogL) > 0 ? (dogW / (dogW + dogL) * 100).toFixed(1) : 'N/A';
  const mark = t === 0.63 ? ' ← current' : '';

  console.log(`${t.toFixed(2).padEnd(7)}| ${String(w).padEnd(5)}| ${String(l).padEnd(5)}| ${pct.padEnd(6)}| ${units.padEnd(8)}| ${perDay.padEnd(10)}| ${favPct.padEnd(7)}| ${dogPct}${mark}`);
}
